src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/Cohost/HtmlDocumentSynchronizer.SynchronizationRequest.cs

```diff
@@ -14,21 +14,22 @@ private class SynchronizationRequest(RazorDocumentVersion requestedVersion) : ID
     {
         private readonly RazorDocumentVersion _requestedVersion = requestedVersion;
         private readonly TaskCompletionSource<bool> _tcs = new();
-        private readonly CancellationTokenSource _cts = new(TimeSpan.FromMinutes(5));
+        private CancellationTokenSource? _cts;
 
         public Task<bool> Task => _tcs.Task;
 
         public RazorDocumentVersion RequestedVersion => _requestedVersion;
 
-        internal static SynchronizationRequest CreateAndStart(TextDocument document, RazorDocumentVersion requestedVersion, Func<TextDocument, CancellationToken, Task> syncFunction)
+        internal static SynchronizationRequest CreateAndStart(TextDocument document, RazorDocumentVersion requestedVersion, Func<TextDocument, CancellationToken, Task<bool>> syncFunction)
         {
             var request = new SynchronizationRequest(requestedVersion);
             request.Start(document, syncFunction);
             return request;
         }
 
-        private void Start(TextDocument document, Func<TextDocument, CancellationToken, Task> syncFunction)
+        private void Start(TextDocument document, Func<TextDocument, CancellationToken, Task<bool>> syncFunction)
         {
+            _cts = new(TimeSpan.FromMinutes(1));
```

Comment:
Is there a reason that the time was reduced?

Reply:
I never meant it to be 5 minutes anyway, I just never cleaned up from initial testing. Given this now uses OOP, I don't expect the timeout will ever be hit anyway, as if OOP is broken, so is the rest of Roslyn, and all bets are probably off.